### PR TITLE
First round of changes to api-reference

### PR DESCRIPTION
**Pull request by NoSQLKnowHow**
- Added dictionary collection API reference
- Converted api-reference to be a section
- Added page for collection response objects
- Added page for collectionTtl objects
- added more links

### CODE REVIEW

File: docs/develop/api-reference/collection-ttl.md
Comment (cprice404) [inline docs/develop/api-reference/collection-ttl.md]: nit: maybe rather than "a little less bewildering" we say "as straightforward as possible" or "more straightforward" or "more intuitive"?

File: docs/develop/api-reference/collection-ttl.md
Comment (cprice404) [inline docs/develop/api-reference/collection-ttl.md]: may be worth also specifying the behavior for the false case?  "Otherwise, the collection's existing TTL will be retained"

File: docs/develop/api-reference/dictionary-collections.md
Comment (cprice404) [inline docs/develop/api-reference/dictionary-collections.md:12]: You asked about this in slack; in the context here I think this reads fine, personally.

File: docs/develop/api-reference/dictionary-collections.md
Comment (cprice404) [inline docs/develop/api-reference/dictionary-collections.md]: Would probably remove the word "item" here as it might be interpreted as only pulling a single field from the dictionary.  Maybe "retrieves an entire dictionary from a cache" or something?

File: docs/develop/api-reference/dictionary-collections.md
Comment (cprice404) [inline docs/develop/api-reference/dictionary-collections.md]: Same comment here, to me the word "item" after "dictionary" is superfluous and possibly confusing.

File: docs/develop/api-reference/dictionary-collections.md
Comment (cprice404) [inline docs/develop/api-reference/dictionary-collections.md:55]: not sure if it might be worth stating that these two return the field name?  if you feel like that is obvious I'm good

File: docs/develop/api-reference/dictionary-collections.md
Comment (cprice404) [inline docs/develop/api-reference/dictionary-collections.md]: same as elsewhere, basically all places where you have "dictionary item" I would probably see how you feel about the text with just "dictionary" instead.

File: docs/develop/api-reference/dictionary-collections.md
Comment (cprice404) [inline docs/develop/api-reference/dictionary-collections.md]: "If the dictionary does not exist, it is created with a single field"

File: docs/develop/api-reference/dictionary-collections.md
Comment (cprice404) [inline docs/develop/api-reference/dictionary-collections.md]: "If the dictionary does not exist, ..."

File: docs/develop/api-reference/list-collections.md
Comment (cprice404) [inline docs/develop/api-reference/list-collections.md]: same comment as for dictionary, IMO the word "item" is unnecessary for these

File: docs/develop/api-reference/list-collections.md
Comment (cprice404) [inline docs/develop/api-reference/list-collections.md:40]: 👍

File: docs/develop/api-reference/response-objects.md
Comment (cprice404) [inline docs/develop/api-reference/response-objects.md]: Not sure if the `extends Response` part adds value here.  That's pretty specific to the individual programming languages.

File: docs/develop/api-reference/response-objects.md
Comment (cprice404) [inline docs/develop/api-reference/response-objects.md]: this definitely should not go in public docs :)

File: docs/develop/api-reference/response-objects.md
Comment (cprice404) [inline docs/develop/api-reference/response-objects.md]: I would remove this entire `### Response` section.  This is internal sausage-making stuff.  Just skip straight to the `Error` part.

File: docs/develop/api-reference/response-objects.md
Comment (cprice404) [inline docs/develop/api-reference/response-objects.md]: remove `extends Response` stuff for all the rest of these too.

**Review by cprice404 — COMMENTED** [review]
This is looking awesome, thanks so much for your hard work!  Left a handful of ideas/thoughts/comments.

File: docs/develop/api-reference/dictionary-collections.md
Comment (NoSQLKnowHow) [reply docs/develop/api-reference/dictionary-collections.md:55]: When you say they return the field name, doesn't this return the entire entity from the dictionary? So the key and value? Maybe I am missing something here.

Comment (cprice404) [reply docs/develop/api-reference/dictionary-collections.md:55]: "Field" is the name of the field from the dictionary.
"Value" is the value of the field from the dictionary.

File: docs/develop/api-reference/dictionary-collections.md
Comment (cprice404) [inline docs/develop/api-reference/dictionary-collections.md]: no apostrophe for `it's` here, I think?

**Review by cprice404 — COMMENTED** [review]
just one more minor typo and then I'm +1.  I do think it'd be worth getting one more quick review from @eaddingtonwhite or @kvcache since this is a foundational PR.

File: docs/develop/api-reference/dictionary-collections.md
Comment (NoSQLKnowHow) [reply docs/develop/api-reference/dictionary-collections.md]: fixed and pushed a commit.

**Review by cprice404 — APPROVED** [review]
lgtm, your call on whether to get one more set of 👀 from @eaddingtonwhite or @kvcache before you merge.

**Pull request merged**
